packages/stellar_client/lib/src/client.dart

```diff
@@ -518,4 +562,363 @@ class Client {
+      {required String sellingAssetCode,
+      required String buyingAssetCode}) async {
+    if (!_currencies.currencies.containsKey(sellingAssetCode)) {
+      throw Exception('Sell asset $sellingAssetCode is not available.');
+    }
+    if (!_currencies.currencies.containsKey(buyingAssetCode)) {
+      throw Exception('Buy asset $buyingAssetCode is not available.');
+    }
+    http.Client httpClient = http.Client();
+    Uri serverURI = Uri.parse(_horizonServerUrls[_network.toString()]!);
+
+    final Asset sellingAsset = _getAsset(sellingAssetCode);
+    final Asset buyingAsset = _getAsset(buyingAssetCode);
+
+    OrderBookRequestBuilder orderBookRequest =
+        OrderBookRequestBuilder(httpClient, serverURI)
+          ..sellingAsset(sellingAsset)
+          ..buyingAsset(buyingAsset);
+
+    return await orderBookRequest.stream();
+  }
+
+  /// Lists all active offers created by the current account.
+  ///
+  /// This function fetches a list of `OfferResponse` objects representing
+  /// open orders created by the account.
+  ///
+  /// ### Understanding Stellar Order Representation:
+  /// - **Price (`OfferResponse.price`)**: Stellar stores price as `buying / selling`,
+  ///   meaning the displayed price is the **inverse** of the price provided
+  ///   when creating an order.
+  /// - **Amount (`OfferResponse.amount`)**: This represents the amount of the
+  ///   **buying asset** still available for trade, not the original amount
+  ///   of the selling asset.
+  ///
+  /// ### Conversion Formula:
+  /// When placing an order:
+  /// ```
+  /// Total selling amount = Buying amount * Price
+  /// ```
+  ///
+  /// Stellar inverts the price when storing the offer:
+  /// ```
+  /// Stored price = 1 / Provided price
+  /// ```
+  ///
+  /// ### Example:
+  /// #### **Creating an Order**
+  /// ```dart
+  /// await stellarClient.createOrder(
+  ///     sellingAssetCode: 'USDC',
+  ///     buyingAssetCode: 'TFT',
+  ///     amount: '5',     // Buying 5 TFT
+  ///     price: '0.02');  // 1 USDC = 0.02 TFT
+  /// ```
+  ///
+  /// #### **Retrieved Offer (from Stellar Order Book)**
+  /// ```dart
+  /// OfferResponse {
+  ///   amount: "0.2",    // Total selling amount = 5 * 0.02 = 0.2 USDC
+  ///   price: "50.0"     // Inverted: 1 / 0.02 = 50 USDC per TFT
+  /// }
+  /// ```
+  ///
+  /// **Key Takeaways:**
+  /// - `OfferResponse.amount` = **Total amount of the selling asset left**.
+  /// - `OfferResponse.price` = **Inverse of the provided price**.
+  Future<List<OfferResponse>> listMyOffers() async {
+    try {
+      final offers = await _sdk.offers.forAccount(accountId).execute();
+
+      if (offers.records.isEmpty) {
+        logger.i('No offers found for account: $accountId');
+        return [];
+      }
+
+      return offers.records;
+    } catch (error) {
+      throw Exception('Error listing offers for account $accountId: $error');
+    }
+  }
+
+  Future<List<TradeResponse>> getTradingHistory(String accountId) async {
```

Comment:
Don't we already have the account ID within the client instance?

Reply:
or move it out the client to be a helper as it doesn't require having a wallet secret